### PR TITLE
Changed info_xml_creator to use std::ostream

### PR DESCRIPTION
**Pull request by npwoods**
Also, we are now passing the stream to every function, rather than sloppily keeping it in a member variable

### CODE REVIEW

**Comment by cuavas** [comment]
I experimented with this before and found it caused a substantial drop in performance.  Did you actually test the performance impact of this?  Also, why are you creating temporary strings all over the place?  That's going to massively hurt performance in itself.

File: src/frontend/mame/info.cpp [hunk 1]
```diff
@@ -352,41 +346,42 @@ void info_xml_creator::output(FILE *out, const std::function<bool(const char *sh
...
 		strreplace(dtd, "__XML_ROOT__", XML_ROOT);
 		strreplace(dtd, "__XML_TOP__", XML_TOP);
 
-		fprintf(m_output, "%s\n\n", dtd.c_str());
+		out << dtd << std::endl << std::endl;
 	}
 
 	// top-level tag
-	fprintf(m_output, "<%s build=\"%s\" debug=\""
+	out << util::string_format("<%s build=\"%s\" debug=\""
```
Comment (cuavas) [inline src/frontend/mame/info.cpp]: Here's an example - you're creating a temporary output stream and std::string here, when you could use `util::steam_format` to output to the stream directly.

File: src/frontend/mame/info.cpp [hunk 2]
```diff
@@ -352,41 +346,42 @@ void info_xml_creator::output(FILE *out, const std::function<bool(const char *sh
...
 			"yes"
 #else
 			"no"
 #endif
-			"\" mameconfig=\"%d\">\n",
+			"\" mameconfig=\"%d\">",
 			XML_ROOT,
 			util::xml::normalize_string(emulator_info::get_build_version()),
 			CONFIG_VERSION);
+	out << std::endl;
```
Comment (cuavas) [inline src/frontend/mame/info.cpp]: You shoudln't use `std::endl` unnecessarily.  It causes a flush as well as outputting a line ending.  You don't want to be flushing the buffers more than necessary when writing out a lot of content like this.

**Comment by npwoods** [comment]
I did some performance testing and didn't notice a change, though admittedly I was not rigorous, nor did I test any platforms other than Windows.  Incidentally this was intended to be a stepping stone to PR#5443, which (at least for me) does yield a 150% performance benefit.

I'll follow up with a change to switch to use util::stream_format(), which in retrospect would have been not just a more performant solution, but a simpler copy and paste job.

**Comment by npwoods** [comment]
I threw together a PR that changes things to use 'util::stream_format(out, ...' instead of 'out << util::string_format(' and to my surprise, I saw a major performance drop.  Running -listxml went from a 30 second operation to a 3.5 minute operation.

I'm extremely surprised to see this.

**Comment by npwoods** [comment]
I'm pushed my changes here:

https://github.com/npwoods/mame/tree/infocpp_stream_format

But because of the dramatic performance penalty, I am not creating a PR.  Rather, I am providing this so others can replicate my findings.

**Pull request merged**